Source/API/CBLReplication.m

```diff
@@ -537,6 +554,11 @@ - (void) bg_updateProgress {
                           serverCert: serverCert];
         cfrelease(serverCert);
     }];
+
+    if (status == kCBLReplicationStopped) {
+        [_stopLock lockWhenCondition: 0];
```

Comment:
I think there's a possible race condition, because the variable _stopLock gets modified on the main thread. For example, if the replicator stops on its own, and the app also calls -stop, then it would be possible for the value of `_stopLock` to be nil on the first call but non-nil on the second, which would cause an exception because it's calling -unlock before -lock...